### PR TITLE
Stanleykywu/confirming compounds

### LINKED ISSUE #103: Window Confirming number of compounds selected
-Optional: Allow them to select a range of chemicals
-show a table that summarized what it has read from the excel sheet - consistent with other softwares
-select compounds to accept or drop 
-number of compounds selected should be displayed

### PR DESCRIPTION
**Pull request by stanleykywu**
Finish adding a confirmation for inputted csv file in the index page before clustering.

### CODE REVIEW

**Review by sarah-witt — COMMENTED** [review]
Could you hide the MPO column for when an MPO isn't given? Right now it's just a column of 0s. Otherwise looks great!!

**Comment by stanleykywu** [comment]
That has been added!

File: app/routes.py [hunk 1]
```diff
@@ -5,52 +5,60 @@
...
+  if request.method == 'POST':
+    try:
+      data = request.get_array(field_name='file')
+      smiles, include_mpo = construct_smiles(data)
+      session['smiles'] = smiles
+      session['include_mpo'] = include_mpo
+
+      if include_mpo:
+        print(smiles)
+        unique_compounds = pd.DataFrame(dict((k, [v.get('mpo', ''), v['label']]) for k, v in smiles.items()), index=['mpo', 'label']).T
```
Comment (sarah-witt) [inline app/routes.py]: I updated the dictionary names in my last PR, so this should be how to get the MPO score (sorry about that!)
```suggestion
        unique_compounds = pd.DataFrame(dict((k, [v.get('property', ''), v['label']]) for k, v in smiles.items()), index=['mpo', 'label']).T
```

File: app/routes.py [hunk 2]
```diff
@@ -5,52 +5,60 @@
...
-def index():
+@app.route('/index', methods=['GET', 'POST'])
+def index():     
+  session.clear()
+  if request.method == 'POST':
+    try:
+      data = request.get_array(field_name='file')
+      smiles, include_mpo = construct_smiles(data)
+      session['smiles'] = smiles
+      session['include_mpo'] = include_mpo
```
Comment (sarah-witt) [inline app/routes.py]:
```suggestion
      session['include_property'] = include_property
```

File: app/routes.py [hunk 3]
```diff
@@ -5,52 +5,60 @@
...
 @app.route('/')
-@app.route('/index')
-def index():
+@app.route('/index', methods=['GET', 'POST'])
+def index():     
+  session.clear()
+  if request.method == 'POST':
+    try:
+      data = request.get_array(field_name='file')
+      smiles, include_mpo = construct_smiles(data)
```
Comment (sarah-witt) [inline app/routes.py]:
```suggestion
      smiles, include_property = construct_smiles(data)
```

File: app/routes.py [hunk 4]
```diff
@@ -5,52 +5,60 @@
...
+      session['smiles'] = smiles
+      session['include_mpo'] = include_mpo
+
+      if include_mpo:
```
Comment (sarah-witt) [inline app/routes.py]:
```suggestion
      if include_property:
```

File: app/routes.py [hunk 5]
```diff
@@ -5,52 +5,60 @@
...
+        print(smiles)
+        unique_compounds = pd.DataFrame(dict((k, [v.get('mpo', ''), v['label']]) for k, v in smiles.items()), index=['mpo', 'label']).T
+      else:
+        unique_compounds = pd.DataFrame(dict((k, [v['label']]) for k, v in smiles.items()), index=['label']).T
+      return render_template('index.html', title='Cheminformatic Analysis', 
+        unique_compounds=unique_compounds.to_html(), num_compounds=len(unique_compounds), smiles=smiles, include_mpo=include_mpo)
```
Comment (sarah-witt) [inline app/routes.py]:
```suggestion
        unique_compounds=unique_compounds.to_html(), num_compounds=len(unique_compounds), smiles=smiles, include_mpo=include_property)
```

File: app/routes.py [hunk 6]
```diff
@@ -5,52 +5,60 @@
...
-        for smile in bad_smiles.values():
-          reasons_for_failure[smile] += 1
-
-        session['reasons_for_failure'] = reasons_for_failure
-        session.changed = True
-    
-    return render_template('pains_verify_and_coefficient_use.html', title='Cheminformatic Analysis', bad_smiles=bad_smiles, num_remaining=session["num_remaining"], num_removed=session["num_removed"], reasons_for_failure=reasons_for_failure, include_property=session['include_property'])
-
+  smiles = session['smiles']
+  include_mpo = session['include_mpo']
```
Comment (sarah-witt) [inline app/routes.py]:
```suggestion
  include_property = session['include_property']
```

File: app/routes.py [hunk 7]
```diff
@@ -5,52 +5,60 @@
...
+  session['all_smiles'] = convert_to_smiles(smiles.copy())
+  #global good_smiles 
+  good_smiles = convert_to_smiles(filter_smiles(pains.get_smiles(inputs), smiles))
+  session['good_smiles'] = good_smiles
+  #global bad_smiles
+  bs = convert_to_smiles(pains.get_bad_smiles(inputs)) 
+  bad_smiles = bs if isinstance(bs, dict) else {}
+  session['bad_smiles'] = bad_smiles
+  #global include_mpo
+  session['include_mpo'] = include_mpo
```
Comment (sarah-witt) [inline app/routes.py]:
```suggestion
  session['include_property'] = include_property
```

File: app/routes.py [hunk 8]
```diff
@@ -5,52 +5,60 @@
...
+
+  #global reasons_for_failure
+  reasons_for_failure = dict.fromkeys(set(bad_smiles.values()), 0)
+  for smile in bad_smiles.values():
+    reasons_for_failure[smile] += 1
+
+  session['reasons_for_failure'] = reasons_for_failure
+  session.changed = True
+  return render_template('pains_verify_and_coefficient_use.html', title='Cheminformatic Analysis', bad_smiles=bad_smiles, reasons_for_failure=reasons_for_failure, include_mpo=session['include_mpo'])
```
Comment (sarah-witt) [inline app/routes.py]:
```suggestion
  return render_template('pains_verify_and_coefficient_use.html', title='Cheminformatic Analysis', bad_smiles=bad_smiles, reasons_for_failure=reasons_for_failure, include_property=session['include_property'])
```

File: app/templates/index.html [hunk 9]
```diff
@@ -9,13 +9,64 @@ <h1> {{ title }}</h1>
...
+      </style>
+      {% if num_compounds %}
+        <table align="center">
+          <tr>
+            <th># of Compounds</th>
+            <th>Include MPO?</th> 
+          </tr>
+          <tr>
+            <td>{{ num_compounds }}</td>
+            <td>{{ include_mpo }}</td>
```
Comment (sarah-witt) [inline app/templates/index.html]:
```suggestion
            <td>{{ include_property }}</td>
```

File: app/routes.py [hunk 10]
```diff
@@ -5,52 +5,59 @@
...
+  session['include_property'] = include_property
+
+  #global reasons_for_failure
+  reasons_for_failure = dict.fromkeys(set(bad_smiles.values()), 0)
+  for smile in bad_smiles.values():
+    reasons_for_failure[smile] += 1
+
+  session['reasons_for_failure'] = reasons_for_failure
+  session.changed = True
+  return render_template('pains_verify_and_coefficient_use.html', title='Cheminformatic Analysis', bad_smiles=bad_smiles, reasons_for_failure=reasons_for_failure, include_property=session['include_property'])
```
Comment (sarah-witt) [inline app/routes.py]:
```suggestion
  return render_template('pains_verify_and_coefficient_use.html', title='Cheminformatic Analysis', bad_smiles=bad_smiles, num_remaining=session["num_remaining"], num_removed=session["num_removed"], reasons_for_failure=reasons_for_failure, include_property=session['include_property'])
```

**Review by sarah-witt — APPROVED** [review]
:tada:

**Pull request merged**